compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -616,8 +619,8 @@ class OMR_EXTENSIBLE CodeGenerator
    bool isMaterialized(TR::Node *); // no virt, cast
    bool isMaterialized(int64_t); // no virt, cast
    bool materializesLargeConstants() { return false;}
-   int32_t getLargestNegConstThatMustBeMaterialized() {return -32769;} // no virt, cast
-   int32_t getSmallestPosConstThatMustBeMaterialized() {return 32768;} // no virt, cast
+   int64_t getLargestNegConstThatMustBeMaterialized() {return -32769;}
+   int64_t getSmallestPosConstThatMustBeMaterialized() {return 32768;}
```

Comment:
My latest commit basically does this: answer NO by default and answer yes in the P and Z layer. And for the platforms that answer yes to `isMaterialized`, they'll provide platform specific ranges using the `SmallestPosConstThatMustBeMaterialized` and `SmallestPosConstThatMustBeMaterialized` queries, which are not provided in the common codegen layer.